doc/changelog/04-tactics/16289-eauto-cost.rst

```diff
@@ -0,0 +1,6 @@
+- **Fixed:**
+  :tacn:`eauto` respects priorities of ``Extern`` hints
```

Comment:
I think that the changelog should be slightly more explicit about the fact that this might break developments or introduce changes of performance profile.

Reply:
Indeed, this should probably be a **Changed** instead of a **Fixed** for this reason, and some information on what kind of breakage to expect should be provided (recall that when users upgrade they get compatibility issues without necessarily knowing what caused them).

Reply:
In addition, it could make sense to add a note of this change of behavior in the documentation of `Hint Extern` using a `.. versionchanged::` directive. We don't frequently do this but here is an example: https://github.com/coq/coq/blob/4e554eac6a822f919beeae4a6f1ce3dfab0d7c8c/doc/sphinx/practical-tools/coq-commands.rst?plain=1#L302-L306

Reply:
I added a `.. warning::` to the changelog, also indicating how to approximate prior behavior, and set the description to `Changed`. To set `.. versionchanged::` does not feel correct. Previously, it was an undocumented quirk of specifically `eauto`, which led unexpected behavior and bugs.